### ISSUE TITLE
binary for OS X is linked dinamically against a specific libGLEW version

### ISSUE THREAD

**Issue by superbonaci** [sm64pc/sm64pc#478]
The 18th June 2020, I compiled sm64ex, for mac OS X latest version at that time, for Intel cpu. Of course I had installed brew and required formulae, everything explained and solved in this issue: https://github.com/sm64pc/sm64ex/issues/26

Today (4th september 2021), I updated to OS X Big Sur 11.5.2, and also all brew formulae. However, the same compiled binary `sm64pc/build/us_pc/sm64.us.f3dex2e` didn't work now straightaway:
````
./sm64.us.f3dex2e   
dyld: Library not loaded: /usr/local/opt/glew/lib/libGLEW.2.1.dylib
  Referenced from: /Users/mcsr/sm64pc/build/us_pc/./sm64.us.f3dex2e
  Reason: image not found
zsh: abort      ./sm64.us.f3dex2e
````
Linking required library to new one seems to work:
````
cd /usr/local/opt/glew/lib
ln -s libGLEW.dylib libGLEW.2.1.dylib
````
This works because `ibGLEW.dylib` is a symlink to `libGLEW.2.2.dylib`, which is currently the latest version of the library.

My conclusion is that `sm64pc` binary was dynamically linked against a specific version of `libGLEW`, which requires to install brew and such library to run in on any new system. Can it be linked dynamically against any version of `libGLEW`, or ideally compiled statically? Didn't find any info in documentation and I don't recall anybody had this issue before.

**Comment by fgsfdsfgs** [comment]
This is something that I cannot solve since I know nothing about the OSX ecosystem and indeed don't own any OSX devices. You'll have to check out the part in the Makefile that specifies what to link in OSX mode.

**Comment by Xenthio** [comment]
> The 18th June 2020, I compiled sm64ex, for mac OS X latest version at that time, for Intel cpu. Of course I had installed brew and required formulae, everything explained and solved in this issue: #26
> 
> Today (4th september 2021), I updated to OS X Big Sur 11.5.2, and also all brew formulae. However, the same compiled binary `sm64pc/build/us_pc/sm64.us.f3dex2e` didn't work now straightaway:
> 
> ```
> ./sm64.us.f3dex2e   
> dyld: Library not loaded: /usr/local/opt/glew/lib/libGLEW.2.1.dylib
>   Referenced from: /Users/mcsr/sm64pc/build/us_pc/./sm64.us.f3dex2e
>   Reason: image not found
> zsh: abort      ./sm64.us.f3dex2e
> ```
> 
> Linking required library to new one seems to work:
> 
> ```
> cd /usr/local/opt/glew/lib
> ln -s libGLEW.dylib libGLEW.2.1.dylib
> ```
> 
> This works because `ibGLEW.dylib` is a symlink to `libGLEW.2.2.dylib`, which is currently the latest version of the library.
> 
> My conclusion is that `sm64pc` binary was dynamically linked against a specific version of `libGLEW`, which requires to install brew and such library to run in on any new system. Can it be linked dynamically against any version of `libGLEW`, or ideally compiled statically? Didn't find any info in documentation and I don't recall anybody had this issue before.

Builds for me just fine on macOS 12.1 with libglew v2.2

**Comment by superbonaci** [comment]
@Xenthio after db9a634 I guess.
